app/assets/javascripts/social_share_button.js

```diff
@@ -0,0 +1,27 @@
+// Opening pop-ups with share URL
+function openShareUrl(url, initialWidth = 640, initialHeight = 480) {
+  if (typeof url !== "string" || !url.startsWith("http")) {
+    console.error("Invalid URL"); // Consider removing this line if console warnings should be avoided.
+    return;
+  }
+
+  const width = Math.max(100, Math.min(screen.width, initialWidth));
+  const height = Math.max(100, Math.min(screen.height, initialHeight));
+
+  const left = (screen.width / 2) - (width / 2);
+  const top = (screen.height * 0.3) - (height / 2);
+  const opts = `width=${width},height=${height},left=${left},top=${top},menubar=no,status=no,location=no`;
+
+  window.open(url, "popup", opts);
+}
+
+$(document).ready(function () {
+  $(".ssb-icon").on("click", function (e) {
```

Comment:
This is not going to work with Turbo paging because `$(document).ready` won't happen when you click *Newer/Older Entries*. This can be fixed by delegating the event to something that stays on the page or the entire document, but the other question is: Should there be share buttons on multiple entry pages at all? We're also trying to truncate the entries on such pages, with #5121 or something like. You made your own version in #5125 and I don't see share buttons in it. Maybe the buttons should only be present on single entry pages?

Reply:
I agree on this one. While working on this, there was not too much information indicating someone is going to do such thing. If we are going to truncate the text, then this definitely should only be displayed on single entry pages. Although I'm not sure should I do it right away or wait for #5121 to be merged. Thank you.

Reply:
It's already not working correctly on index pages so you'll have to do something. Removing the buttons is easier than fixing them, right?